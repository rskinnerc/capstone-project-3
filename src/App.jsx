import { Route, Routes } from 'react-router-dom';
import Home from './pages/Home';
import Genre from './pages/Genre';
import './App.css';
import Navbar from './components/Navbar';

function App() {
  return (
    <div>
      <Navbar />
      <Routes>
        <Route index element={<Home />} />
        <Route path="/genre/:genreSlug" element={<Genre />} />
      </Routes>
    </div>
  );
}

export default App;
