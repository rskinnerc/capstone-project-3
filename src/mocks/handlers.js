import { rest } from 'msw';

const handlers = [
  rest.get('https://api.rawg.io/api/genres', (req, res, ctx) => {
    return res(
      ctx.json(
        {
          "count": 19, "next": null, "previous": null, "results": [{ "id": 4, "name": "Action", "slug": "action", "games_count": 153243, "image_background": "https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg" }]
        })
    );
  }),
  rest.get('https://api.rawg.io/api/genres/action', (req, res, ctx) => {
    return res(
      ctx.json({ "id": 4, "name": "Action", "slug": "action", "games_count": 153243, "image_background": "https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg" })
    )
  }),
  rest.get('https://api.rawg.io/api/games', (req, res, ctx) => {
    return res(
      ctx.json({
        "count": 153246,
        "previous": null,
        "results": [
          {
            "slug": "grand-theft-auto-v",
            "name": "Grand Theft Auto V",
            "rating": 4.48,
            "ratings_count": 5720,
            "id": 3498,
          }
        ]
      })
    )
  }),
];

export default handlers;
