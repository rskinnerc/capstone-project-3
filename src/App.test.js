import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import App from './App';
import store from './redux/configureStore';
import { server } from './mocks/server';

describe('The full App with routes', () => {
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('should render the home page', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>,
    );

    expect(screen.queryByText('The Games Database')).toBeInTheDocument();
    expect(screen.queryByText('STATS BY GENRE')).toBeInTheDocument();
  });

  it('should render the genre page', async () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/genre/action']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>,
    );

    const breakdownText = await screen.findByText('ACTION GAMES BREAKDOWN', { exact: false });
    const countText = await screen.findByText('Games Count', { exact: false });

    expect(breakdownText).toBeInTheDocument();
    expect(countText).toBeInTheDocument();
  });

  it('should get back from the genre page to the home page with the back button', async () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/genre/action']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>,
    );

    const backButton = await screen.findByTestId('backButton');
    userEvent.click(backButton);

    const headingText = await screen.findByText('The largest video games database.');
    expect(headingText).toBeInTheDocument();
  });

  it('should maintain the snapshot btween renders', async () => {
    const dom = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>,
    );

    await waitFor(() => expect(dom).toMatchSnapshot());
  });
});
