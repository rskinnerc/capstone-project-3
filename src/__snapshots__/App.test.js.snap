// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The full App with routes should maintain the snapshot btween renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-gradient-to-r from-marino-900 to-marino-900 via-marino-700 min-h-screen"
      >
        <nav
          class="flex flex-row justify-items-center bg-marino-600 text-white py-2 sm:w-96 mx-auto"
        >
          <h1
            class="basis-10/12 pl-3 font-semibold font-gill"
          >
             
            The Games Database
          </h1>
          <i
            class="bi bi-mic-fill justify-self-end basis-1/12"
          />
          <i
            class="bi bi-gear-fill justify-self-end basis-1/12"
          />
        </nav>
        <div
          class="sm:w-96 mx-auto"
        >
          <div
            class="bg-gradient-to-br from-marino-500 to-marino-900 flex flex-col items-center text-white px-5 py-8"
          >
            <h1
              class="text-2xl text-center font-lato"
            >
              The largest video games database.
            </h1>
            <p
              class="my-6 text-center"
            >
              More than
               
              <span
                class="font-bold text-marino-200 text-xl"
              >
                350.000
              </span>
               
              games for 50 platforms including mobiles. Start now filtering by genre!
            </p>
            <input
              class="bg-marino-50 rounded-md p-3 border-0 outline-0 ring-2 focus:ring-marino-400 ring-marino-200 text-marino-600 font-bold"
              placeholder="Filter by genre"
              type="text"
              value=""
            />
          </div>
          <div
            class="font-lato text-sm bg-marino-600 text-white p-1"
          >
            STATS BY GENRE
          </div>
          <ul
            class="flex flex-row flex-wrap"
          >
            <li
              class="odd:bg-marino-500 h-52 even:bg-marino-700 basis-1/2 relative"
            >
              <a
                href="/genre/action"
              >
                <i
                  class="bi bi-arrow-right-circle absolute top-0.5 right-1 z-30 font-bold text-white"
                />
                <img
                  alt="Action"
                  class="h-4/5 grayscale mix-blend-luminosity hover:grayscale-0 hover:mix-blend-normal"
                  src="https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
                />
                <h2
                  class="font-lato font-semibold text-white px-1 text-right"
                >
                  Action
                </h2>
                <h3
                  class="text-sm text-white font-lato px-1 text-right"
                >
                  <span
                    class="text-marino-200"
                  >
                    Games Count:
                  </span>
                   
                  153243
                </h3>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-gradient-to-r from-marino-900 to-marino-900 via-marino-700 min-h-screen"
    >
      <nav
        class="flex flex-row justify-items-center bg-marino-600 text-white py-2 sm:w-96 mx-auto"
      >
        <h1
          class="basis-10/12 pl-3 font-semibold font-gill"
        >
           
          The Games Database
        </h1>
        <i
          class="bi bi-mic-fill justify-self-end basis-1/12"
        />
        <i
          class="bi bi-gear-fill justify-self-end basis-1/12"
        />
      </nav>
      <div
        class="sm:w-96 mx-auto"
      >
        <div
          class="bg-gradient-to-br from-marino-500 to-marino-900 flex flex-col items-center text-white px-5 py-8"
        >
          <h1
            class="text-2xl text-center font-lato"
          >
            The largest video games database.
          </h1>
          <p
            class="my-6 text-center"
          >
            More than
             
            <span
              class="font-bold text-marino-200 text-xl"
            >
              350.000
            </span>
             
            games for 50 platforms including mobiles. Start now filtering by genre!
          </p>
          <input
            class="bg-marino-50 rounded-md p-3 border-0 outline-0 ring-2 focus:ring-marino-400 ring-marino-200 text-marino-600 font-bold"
            placeholder="Filter by genre"
            type="text"
            value=""
          />
        </div>
        <div
          class="font-lato text-sm bg-marino-600 text-white p-1"
        >
          STATS BY GENRE
        </div>
        <ul
          class="flex flex-row flex-wrap"
        >
          <li
            class="odd:bg-marino-500 h-52 even:bg-marino-700 basis-1/2 relative"
          >
            <a
              href="/genre/action"
            >
              <i
                class="bi bi-arrow-right-circle absolute top-0.5 right-1 z-30 font-bold text-white"
              />
              <img
                alt="Action"
                class="h-4/5 grayscale mix-blend-luminosity hover:grayscale-0 hover:mix-blend-normal"
                src="https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg"
              />
              <h2
                class="font-lato font-semibold text-white px-1 text-right"
              >
                Action
              </h2>
              <h3
                class="text-sm text-white font-lato px-1 text-right"
              >
                <span
                  class="text-marino-200"
                >
                  Games Count:
                </span>
                 
                153243
              </h3>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
